core/src/main/java/org/elasticsearch/rest/action/RestActions.java

```diff
@@ -69,15 +69,23 @@ public static void buildBroadcastShardsHeader(XContentBuilder builder, Params pa
                                    response.getShardFailures());
     }
 
+    public static final class ShardsFields {
+        public static final String _SHARDS = "_shards";
+        public static final String TOTAL = "total";
+        public static final String SUCCESSFUL = "successful";
+        public static final String FAILED = "failed";
+        public static final String FAILURES = "failures";
+    }
+
     public static void buildBroadcastShardsHeader(XContentBuilder builder, Params params,
                                                   int total, int successful, int failed,
                                                   ShardOperationFailedException[] shardFailures) throws IOException {
-        builder.startObject("_shards");
-        builder.field("total", total);
-        builder.field("successful", successful);
-        builder.field("failed", failed);
+        builder.startObject(ShardsFields._SHARDS);
```

Comment:
This is a good idea, I started looking into it but it looks like this needs quiet a bit of moving code around, so I'd prefer to do that in a separate PR if you're okay with that.